cheroot/__main__.py

```diff
@@ -114,20 +136,25 @@ def main(args=None):
     if chdir not in sys.path:
         sys.path.insert(0, chdir)
 
+    # validate arguments
+    wsgi_app = get_wsgi_app(raw_args._wsgi_app)
+    addr, port = parse_wsgi_bind_addr(raw_args._bind_addr)
+    if port is None:
+        bind_addr = addr
+    else:
+        bind_addr = addr, port
+
     # create a Server object based on the arguments provided
     server_args = {
-        k: v for k, v in vars(raw_args).items()
-        if (not k.startswith('_')) and v is not None
+        'wsgi_app': wsgi_app,
+        'bind_addr': bind_addr,
     }
-    server_args['wsgi_app'] = wsgi_app_path(raw_args._wsgi_app)
+    for arg, value in vars(raw_args).items():
```

Comment:
It's still too magical as it prevents code analyzers from identifying problems early and breaks debugging experience.

_[Explicit is better than implicit. (c) Zen Of Python](https://zen-of-python.info/explicit-is-better-than-implicit.html#2)_